packages/slides-frontend/src/utils/helperFunctions.js

```diff
@@ -0,0 +1,8 @@
+// check if the image name string appears in state
+// if only one, it means that it is the last time the image appears in the slides and has to be deleted
+
+const deleteImageFromBackend = (imageName, slidesStringified) => (slidesStringified.match(new RegExp(imageName,"g")) || []).length === 1;
```

Comment:
```suggestion
const deleteImageFromBackend = (imageName, slidesStringified) => (slidesStringified.match(new RegExp(imageName, "g")) || []).length === 1;
```

Reply:
Hm.. that might not be the best way of checking if there are more than 1 images with the same name. What if e.g. I add the text element with the image name as the content.

Reply:
Well, the text is saved as html in the redux store, in order to keep the styles.
So even if there are no styles, the text will have `<p>` tags around it.
A workaround would be, instead of checking for the `imageName` string, to search for the  `"type":"IMAGE","Src":"${imageName}"` string. This way even if the user has written the same text in a text box, in the stringified state it will be like: 
```
"type":"TEXT","Data":"<p>\"type\":\"IMAGE\",\"Src\":\"5ae3c763551cbea7ec39302742ef8455_me.JPG\"</p>\n"
```
which doesn't match with the image name, so it doesn't get confused.
I tried it and works, is it good enough?

Reply:
Don't you think it is a bit overcomplicated? Why don't you just iterate over all images elements and check if their name property is the same?

Reply:
ok so get the state as an object and have a loop?
I think it will be more complicated but okay I can do it no problem

Reply:
I think it will be better in the long run and much easier to understand in the future.

Reply:
``` js
const imageArray = slides.map(slide =>
        slide.itemsArray.map(item => {
            if (item.type === "IMAGE" && item.Src === imageName) {
                return item.Src;
            }
        })
    );
```

this doesn't give me the expected results... Any idea?

Reply:
ok did it

Reply:
we have to consider what happens in the remove slide, when the pictures should be deleted from backend.

Reply:
Right now, if a slide has a picture and is the last one that has it, the picture file doesn't get deleted in the backend.